editoast/src/models/paced_train.rs

```diff
@@ -284,53 +324,69 @@ impl From<PacedTrain> for paced_train::PacedTrain {
                     .map(|main_category| TrainCategory::main(main_category.0))
                     .xor(paced_train.sub_category.map(TrainCategory::sub)),
             },
-            paced: Paced {
-                time_window: paced_train.time_window.try_into().unwrap(),
-                interval: paced_train.interval.try_into().unwrap(),
-                exceptions: paced_train.exceptions,
-            },
+            paced: paced_train.time_window.and_then(|time_window| {
+                paced_train.interval.map(|interval| Paced {
+                    time_window: time_window.try_into().unwrap(),
+                    interval: interval.try_into().unwrap(),
+                    exceptions: paced_train.exceptions,
+                })
+            }),
         }
     }
 }
 
-#[derive(Debug, Clone, Hash, Serialize, Deserialize, PartialEq, ToSchema)]
-#[serde(tag = "type")]
-pub enum OccurrenceId {
-    BaseOccurrence { index: u64 },
-    ModifiedException { index: u64, exception_key: String },
-    CreatedException { exception_key: String },
-}
-
-#[derive(Debug, Clone, Hash, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema, Hash)]
+#[serde(tag = "type", rename_all = "snake_case")]
 /// This ID is used to identify paced train occurrences and exceptions when sending them to the core API for conflict detection.
 pub enum TrainId {
-    TrainSchedule(i64),
-    PacedTrain {
-        paced_train_id: i64,
-        occurrence_id: OccurrenceId,
+    Single {
+        id: i64,
+    },
+    Occurrence {
+        id: i64,
+        index: u64,
+        exception_key: Option<String>,
+    },
+    CreatedOccurrence {
+        id: i64,
+        exception_key: String,
     },
 }
```

Comment:
This need to be discussed. I unified `OccurrenceId` and `TrainId`.
I opted for three variants, but I could also go for four like:

```rust
pub enum TrainId {
    Single {
        id: i64,
    },
    BaseOccurrence {
        id: i64,
        index: u64,
    },
    ModifiedOccurrence {
        id: i64,
        index: u64.
        exception_key: String,
    },
    CreatedOccurrence {
        id: i64,
        exception_key: String,
    },
Comment view}
```

Reply:
I would go for 4 variants, like it was done before, i.e., the choice to separate a “regular occurrence” from a “modified occurrence”. I find it clearer to have one `enum` with 4 variants, than an `enum` with 3 variants, for which one of them have a sub-variant (`Option`) which is semantically less clear (`Option` can mean anything, it’s a generic type, whereas `BaseOccurrence`[^1] and `ModifiedOccurrence` are distinguishable in our business jargon).

[^1]:  or just `Occurrence` , I’m fine with a shorter name

Reply:
Good for me (just a small comment on the naming). I’ll let you resolve since you initiate the thread.